packages/lib/getAggregatedAvailability.ts

```diff
@@ -44,10 +45,28 @@ export const getAggregatedAvailability = (
   const dateRangesToIntersect = !!fixedDateRanges.length ? [fixedDateRanges] : [];
   const roundRobinHosts = userAvailability.filter(({ user }) => user?.isFixed !== true);
   if (roundRobinHosts.length) {
-    dateRangesToIntersect.push(
-      roundRobinHosts.flatMap((s) => (!isTeamEvent ? s.dateRanges : s.oooExcludedDateRanges))
-    );
+    // Group round robin hosts by their groupId
+    const hostsByGroup: Record<string, typeof roundRobinHosts> = {};
+
+    roundRobinHosts.forEach((host) => {
+      const groupId = host.user?.groupId || DEFAULT_GROUP_ID;
+      if (!hostsByGroup[groupId]) {
+        hostsByGroup[groupId] = [];
+      }
+      hostsByGroup[groupId].push(host);
+    });
```

Comment:
```suggestion
  const hostsByGroup = roundRobinHosts.reduce((groups, host) => {
    const groupId = host.user?.groupId || DEFAULT_GROUP_ID;
    (groups[groupId] ||= []).push(host);
    return groups;
  }, {} as Record<string, typeof roundRobinHosts>);
```

NIT: I think we can get some better performance here by using a reduce.

Reply:
nit: I don't think there's any performance gain. With N being `roundRobitHosts.length`, both versions are just O(N). But the suggested one reads better, though.